feat(curriculum): create testing quiz (#56610)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-testing/66f1aeffc5774ac692112a7e.md

# --description--

> Answer all of the questions below correctly to pass the quiz on software testing concepts and practices.

Reviewer: ```suggestion
Answer all of the questions below correctly to pass the quiz.
```

this doesn't need to be changed

#### --text--

What type of testing focuses on verifying individual components or functions of code in isolation?

#### --distractors--

> 'Functional testing'

Reviewer 2: ```suggestion
Functional testing
```
Quotes are not needed around distractors or answer. Please remove them for all cases.

To test the system under extreme stress

---

To check for security vulnerabilities

---

To ensure all features are working perfectly

#### --answer--

To verify that the most critical functions of a system are working

### --question--

#### --text--

Which testing practice involves deliberately introducing faults into a system to test its fault-tolerance capabilities?

#### --distractors--

Regression testing

---

Load testing

---

Usability testing

#### --answer--

Chaos testing

### --question--

#### --text--

What does 'TDD' stand for in software development methodologies?

#### --distractors--

Test-Driven Deployment

---

Total Design Development

---

Technical Design Document

#### --answer--

Test-Driven Development

### --question--

#### --text--

Which of the following is an example of a 'non-functional' testing type?

#### --distractors--

Unit testing

---

Integration testing

---

User acceptance testing

#### --answer--

Performance testing

### --question--

#### --text--

Which external CI/CD tool is commonly used for automating builds, tests, and deployments?

#### --distractors--

Jenkins

---

Docker

---

Kubernetes

#### --answer--

> CircleCI

Reviewer: ```suggestion
CircleCI

```

you have accidentally removed a line here

#### --distractors--

Usability testing

---

Compatibility testing

---

Performance testing

#### --answer--

Security testing

### --question--

#### --text--

What type of testing ensures that a system can be used effectively by people with disabilities?

#### --distractors--

Compatibility testing

---

Performance testing

---

Functional testing

#### --answer--

Accessibility testing

### --question--

#### --text--

Which popular JavaScript testing framework is often used for unit testing React applications?

#### --distractors--

Selenium

---

Cypress

---

Playwright

#### --answer--

Jest

### --question--

#### --text--

What term describes the practice of testing an application across different web browsers to ensure consistent functionality?

#### --distractors--

Multi-platform testing

---

Compatibility testing

---

Integration testing

#### --answer--

Cross-browser testing

### --question--

#### --text--

Which CI/CD practice involves automatically running tests whenever code changes are pushed to a repository?

#### --distractors--

Manual testing

---

Regression testing

---

Load testing

#### --answer--

> Automated tests on Github

Reviewer 3: This question is bugging me for two reasons. Regression testing does seem like a valid answer since it is running tests to ensure code changes did not break existing functionality. 

The second reason is that it's technically automated testing being ran in GitHub Actions workflow, but the prompt did specifically say "Automated tests on Github". 

Source for Regression testing : https://www.geeksforgeeks.org/software-engineering-regression-testing/ 

Source on building a Github Actions workflow for testing: https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-net

Reviewer 2: I agree question feels a bit off. It ask about _practice involving automatically running tests_ and answer is _Automated tests_.

Author: About the first concern, the question is specifically asking about the practice of automatically running tests when code changes are pushed, which is a broader concept than just regression testing as for the second one I agree that the answer provided is very specific.  I think switching the answer to Continuous Integration or just 'Automated tests'.

Or another thing we can do is change the question a bit to "Which CI/CD practice involves automatically running tests whenever code changes are pushed to a repository?"

Please lemme know which would be the better option.

Reviewer 3: Let's see. 

"Which CI/CD practice involves automatically running tests whenever code changes are pushed to a repository?" 

A proper answer might be something like "Continuous Testing". 

Source: https://aws.amazon.com/what-is/continuous-testing/

### --question--

#### --text--

What type of testing evaluates the entire system from start to finish, simulating real-world scenarios?

#### --distractors--

Unit testing

---

Integration testing

---

Functional testing

#### --answer--

End-to-end (E2E) testing

### --question--

#### --text--

Which testing approach focuses on how easy and satisfying a product is to use?

#### --distractors--

Performance testing

---

Security testing

---

Compatibility testing

#### --answer--

Usability testing

### --question--

#### --text--

> What does 'CI' stand for in the context of software development practices?

Reviewer 4: ```suggestion
What does CI stand for in the context of software development practices?
```

#### --distractors--

Code Integration

---

Continuous Implementation

---

Centralized Integration

#### --answer--

Continuous Integration

### --question--

#### --text--

Which testing tool is specifically designed for end-to-end testing of web applications across multiple browsers?

#### --distractors--

Jest

---

Mocha

---

Jasmine

#### --answer--

Playwright

### --question--

#### --text--

What type of testing ensures that a system works correctly with different hardware, operating systems, or network environments?

#### --distractors--

Functional testing

---

Unit testing

---

Usability testing

#### --answer--

Compatibility testing

### --question--

#### --text--

> In the context of web accessibility, what does the acronym 'WCAG' stand for?

Reviewer 4: ```suggestion
In the context of web accessibility, what does the acronym WCAG stand for?
```

#### --distractors--

Web Compliance and Accessibility Guidelines

---

World Computer Accessibility Group

---

Web Content Accessibility Grading

#### --answer--

Web Content Accessibility Guidelines

### --question--

#### --text--

> Which of the following is NOT typically considered a part of 'shift-left' testing practices?

Reviewer 4: ```suggestion
Which of the following is NOT typically considered a part of shift-left testing practices?
```

#### --distractors--

Early integration of testing in the development process

---

Increased automation of tests

---

Involving testers from the beginning of the project

#### --answer--

Performing all tests after the development phase

### --question--

#### --text--

> What is the primary purpose of 'smoke testing' in software development?

Reviewer 4: ```suggestion
What is the primary purpose of smoke testing in software development?
```

#### --distractors--

To test the system under extreme stress

---

To check for security vulnerabilities

---

To ensure all features are working perfectly

#### --answer--

To verify that the most critical functions of a system are working

### --question--

#### --text--

Which testing practice involves deliberately introducing faults into a system to test its fault-tolerance capabilities?

#### --distractors--

Regression testing

---

Load testing

---

Usability testing

#### --answer--

Chaos testing

### --question--

#### --text--

> What does 'TDD' stand for in software development methodologies?

Reviewer 4: ```suggestion
What does TDD stand for in software development methodologies?
```

#### --distractors--

Test-Driven Deployment

---

Total Design Development

---

Technical Design Document

#### --answer--

Test-Driven Development

### --question--

#### --text--

> Which of the following is an example of a 'non-functional' testing type?

Reviewer 4: ```suggestion
Which of the following is an example of a non-functional testing type?
```

#### --distractors--

Unit testing

---

Integration testing

---

User acceptance testing

#### --answer--

Performance testing

### --question--

#### --text--

Which external CI/CD tool is commonly used for automating builds, tests, and deployments?

#### --distractors--

> Jenkins

Reviewer 3: I've seen Jenkins in action and it does certainly does allow builds to be automated , tested, and deployed.

Author: Ok I will change it
